test/functional/load_esnext.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+function loadTests() {
+  const fs = require('fs');
+  const path = require('path');
+
+  const directory = path.resolve.apply(path, arguments);
+  fs
+    .readdirSync(directory)
+    .filter(filePath => filePath.match(/.*\.js$/))
+    .map(filePath => path.resolve(directory, filePath))
+    .forEach(x => require(x));
+}
+
+describe('ES2017', function() {
+  let supportES2017 = false;
+  try {
+    new Function('return (async function foo() {})();')();
+    supportES2017 = true;
+  } catch (e) {
+    supportES2017 = false;
+  }
+
+  if (supportES2017) {
+    loadTests(__dirname, '..', 'examples');
+  } else {
+    it.skip('skipping ES2017 tests due to insufficient node version', function() {});
+  }
+});
+
+describe('ES2018', function() {
+  const supportES2018 = !!Symbol.asyncIterator;
```

Comment:
Why do this instead of checking the node version? Every node version above 8 should support async/await for example

Reply:
We could do either, but checking the node version directly is a little harder w/out requiring in `semver`. Doing a quick feature check here avoids that.